espnet/nets/pytorch_backend/transformer/decoder.py

```diff
@@ -115,30 +115,42 @@ def forward(self, tgt, tgt_mask, memory, memory_mask):
             x = self.output_layer(x)
         return x, tgt_mask
 
-    def recognize(self, tgt, tgt_mask, memory):
+    def recognize(self, tgt, tgt_mask, memory, cache=None):
         """Recognize one step.
 
         :param torch.Tensor tgt: input token ids, int64 (batch, maxlen_out)
         :param torch.Tensor tgt_mask: input token mask, uint8  (batch, maxlen_out)
         :param torch.Tensor memory: encoded memory, float32  (batch, maxlen_in, feat)
+        :param List[torch.Tensor] cache: cached output list of (batch, max_time_out-1, size)
         :return x: decoded token score before softmax (batch, maxlen_out, token)
         :rtype: torch.Tensor
```

Comment:
Could you update return value info?

Reply:
OK. I think the return values should be always `y, cache` instead of `y` or `y, cache`. And I will rename `recognize` to `forward_one_step`.